aea/multiplexer.py

```diff
@@ -506,12 +506,19 @@ async def _receiving_loop(self) -> None:
 
                 # process completed receiving tasks.
                 for task in done:
+                    connection = task_to_connection.pop(task)
                     envelope = task.result()
                     if envelope is not None:
+                        if not envelope.is_component_to_component_message:
+                            if envelope.context is None:
+                                envelope.context = EnvelopeContext(
+                                    connection_id=connection.public_id
+                                )
+                            else:
+                                envelope.context.connection_id = connection.public_id
```

Comment:
The framework sets the connection id so the skill knows where the envelope comes `via` in agent to agent communication.